doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,255 @@
+# Average over time
+# =================
+# First, we want to detect those dark spots where the data are lost. In
+# technical terms, we want to *segment* the dark spots (for
+# all frames in the sequence). Unlike the actual data (signal), the dark spots
+# do not move from one frame to the next; they are still. Therefore, we begin
+# by computing the time average of the image sequence. We shall use this
+# time-averaged image to segment the dark spots, the latter then standing out
+# with respect to the background (blurred signal).
+
+image_avg = np.mean(image_seq, axis=0)
+
+print(f'shape: {image_avg.shape}')
+
+fig, ax = plt.subplots(figsize=(6, 6))
+
+ax.imshow(image_avg, cmap='gray')
+ax.set_title('Time-averaged image')
+
+fig.tight_layout()
+
+#####################################################################
+# Use local thresholding
+# ======================
+# To segment the dark spots, we use thresholding. The images we are working
+# with have uneven illumination, which causes variations in the (absolute)
+# intensities of the foreground and the background, from one region to another
+# (distant) one. It is therefore more fitting to compute different threshold
+# values across the image, one for each region. This is called adaptive (or
+# local) thresholding, as opposed to the usual thresholding procedure which
+# employs a single (global) threshold for all pixels in the image.
+#
+# When calling the ``threshold_local`` function from the ``filters`` module,
+# we may change the default neighborhood size (``block_size``), i.e., the
+# typical size (number of pixels) over which illumination varies,
+# as well as the ``offset`` (shifting the neighborhood's weighted mean).
+# Let us try two different values for ``block_size``:
+
+thresh_1 = filters.threshold_local(image_avg, block_size=21, offset=15)
+thresh_2 = filters.threshold_local(image_avg, block_size=43, offset=15)
+
+mask_1 = image_avg < thresh_1
+mask_2 = image_avg < thresh_2
+
+#####################################################################
+# Let us define a convenience function to display two plots side by side, so
+# it is easier for us to compare them:
+
+def plot_comparison(plot1, plot2, title1, title2):
+    fig, (ax1, ax2) = plt.subplots(
+        ncols=2,
+        figsize=(12, 6),
+        sharex=True,
+        sharey=True
+    )
+    ax1.imshow(plot1, cmap='gray')
+    ax1.set_title(title1)
+    ax2.imshow(plot2, cmap='gray')
+    ax2.set_title(title2)
+
+plot_comparison(mask_1, mask_2, "block_size = 21", "block_size = 43")
+
+#####################################################################
+# The "dark spots" appear to be more distinct in the second mask, i.e., the
+# one resulting from using the larger ``block_size`` value.
+# Regarding the value of the offset parameter, we noticed that increasing it
+# instead of keeping its default zero value, would yield a more uniform
+# background, hence letting the objects of interest stand out more visibly.
```

Comment:
I think that these subtle parameter changes could be more generalised. Instead of just explaining why exactly we are changing this specific parameter, maybe we could say "We noticed that changing <blank> parameter caused <blank>. Note: Toggling various parameter values can give us a deeper understanding of the method being used, and can often move us closer to the desired results."

Reply:
Here is what I meant: 
```suggestion
# We noticed that increasing the value of the offset parameter from
# its default zero value would yield a more uniform background,
# hence, letting the objects of interest stand out more visibly. *Note:
# Toggling various parameter values can give us a deeper
# understanding of the method being used, and can often move us
# closer to the desired results.*
```

Reply:
Thanks! I've included the suggestion manually (37d006eae5f3) because I think the note's style should be regular. I think that you should surround 'hence' by commas when you write "A happens *and, hence,* B happens" (in your sentence, the consequence is expressed by the gerund).